### PR TITLE
Thchang/2121 matching generated moment images toggle

### LINKED ISSUE #2121: auto match generated moment images
feature request from @jott3077 
it would be good to provide an option in the moment map generator so that the generated moment images are appended and "matched".

### PR DESCRIPTION
**Pull request by TienHao**
**Description**

Closes #2121.

1. Added a toggle in the moment generator to control the generated moment image(s) to be matched to the spatial reference image.

2. The toggle only shows up when the data source image is the spatial reference image.

3. The controlling property `momentToMatch` was added to AppStore.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] ~e2e test passing~ / corresponding fix added
- [x] changelog updated / ~no changelog update needed~
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] `BackendService` unchanged / ~`BackendService` changed and corresponding ICD test fix added~__

### CODE REVIEW

**Comment by kswang1029** [comment]
@TienHao do you think it is possible to beautify the UI like the following? This would save some space.

Screenshot 2023-08-02 at 11 13 55

**Comment by TienHao** [comment]
@kswang1029 Sure, just changed like this.
截圖 2023-08-02 下午5 11 03

**Review by kswang1029 — APPROVED** [review]
looks good! No regression is seen from e2e tests. Moment related tests are revised.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 * Added a shortcut button for image annotation ([#2167](https://github.com/CARTAvis/carta-frontend/issues/2167)).
 * Added support for AIPS beam images ([#2164](https://github.com/CARTAvis/carta-frontend/issues/2164)).
+* Added a toggle in the moment generator to match generated image(s) ([#2121](https://github.com/CARTAvis/carta-frontend/issues/2167)).
```
Comment (YuHsuan-Hwang) [inline CHANGELOG.md]: Incorrect link.

**Review by YuHsuan-Hwang — COMMENTED** [review]
Looks good. 👍

**Review by crocka — APPROVED** [review]

**Review by YuHsuan-Hwang — APPROVED** [review]

**Pull request merged**